psl-core/src/main/java/org/linqs/psl/util/VizDataCollection.java

```diff
@@ -0,0 +1,233 @@
+                vizData.rules.put(stringRuleId, newRuleElement);
+            }
+        }
+     }
+
+    public static void totalRuleSatDis(List<Rule> rules, GroundRuleStore groundRuleStore) {
+        for (Rule rule : rules) {
+            Iterable<GroundRule> groundedRuleList = groundRuleStore.getGroundRules(rule);
+            double totalSat = 0.0;
+            double totalDis = 0.0;
+            int groundRuleCount = 0;
+            JSONObject moduleElement = new JSONObject();
+
+            for (GroundRule groundRule : groundedRuleList) {
+                if (groundRule instanceof WeightedGroundRule) {
+                    WeightedGroundRule weightedGroundRule = (WeightedGroundRule)groundRule;
+                    totalSat += 1.0 - weightedGroundRule.getIncompatibility();
+                    totalDis += weightedGroundRule.getIncompatibility();
+                }
+                groundRuleCount++;
+            }
+            moduleElement.put("Rule", rule.getName());
+            moduleElement.put("Total Satisfaction", totalSat);
+            moduleElement.put("Satisfaction Percentage", totalSat / groundRuleCount);
+            moduleElement.put("Total Dissatisfaction", totalDis);
+            moduleElement.put("Dissatisfaction Percentage", totalDis / groundRuleCount);
+            vizData.totRuleSatArray.put(moduleElement);
+        }
+    }
+
+    public static void violatedGroundRules(List<Rule> rules, GroundRuleStore groundRuleStore) {
+        for (Rule rule : rules) {
+            JSONObject moduleElement = new JSONObject();
+            double violation = 0.0;
+            boolean weightFlag = false;
+            Iterable<GroundRule> groundedRuleList = groundRuleStore.getGroundRules(rule);
+            for (GroundRule groundRule : groundedRuleList) {
+                if (vizData.violatedGroundRulesList.contains(groundRule)) {
+                    //There can't be weighted violated rules so we can make an assumption here
+                    UnweightedGroundRule unweightedGroundRule = (UnweightedGroundRule)groundRule;
+                    violation = unweightedGroundRule.getInfeasibility();
+                    moduleElement.put("Violated Rule", groundRule.baseToString());
+                    moduleElement.put("Parent Rule", rule.getName());
+                    // moduleElement.put("Weighted", weightFlag);
+                    moduleElement.put("Violation", violation);
+                    vizData.violatedGroundRulesArray.put(moduleElement);
+                }
+            }
+        }
+    }
+
+    public static void ruleMapInsertElement(AbstractLogicalRule parentRule, GroundRule groundRule,
+                            Map<Variable, Integer> variableMap,  Constant[] constantsList) {
+      //Adds a groundAtom element to RuleMap
+      //Why are some groundRules null? Perhaps a thread thing??
+      ArrayList<Integer> atomHashList = new ArrayList<>();
+      if (groundRule != null) {
```

Comment:
Invert this.
Instead of checking if something exists and then having a big `if` around that case,
check if the thing does not exist and bail.

```
if (groundRule == null) {
   return;
}
```

Also, later on in this method you use `groundRule` without checking for it being null.
Inverting this conditional solves that issue as well.

Reply:
Resolved.